src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -227,32 +245,62 @@ export const Map: React.FC<PropsType> = ({
       )),
     [attendances, setIsRoomModalOpen, setSelectedRoom, venue]
   );
+
+  const zoomIn = useCallback(() => {
+    setZoom((zoom) => Math.min(zoom + ZOOM_INCREMENT, MAX_ZOOM));
+  }, []);
+
+  const zoomOut = useCallback(() => {
+    setZoom((zoom) => Math.max(zoom - ZOOM_INCREMENT, MIN_ZOOM));
+  }, []);
+
   if (!user || !venue) {
     return <>Loading map...</>;
   }
 
+  const translateX =
+    (((mapSize.width * zoom - (mapSize.width * zoom) / zoom) / zoom) * 2) / 4;
+  const translateY =
+    (((mapSize.height * zoom - (mapSize.height * zoom) / zoom) / zoom) * 2) / 4;
+  const bgRatio = mapSize.width ? mapSize.width / mapSize.height : 1;
+
+  const previewAvatarPosition = {
+    x: currentPosition?.row ? (currentPosition?.row / rows) * 100 - 2 : 0,
+    y: currentPosition?.column
+      ? (currentPosition?.column / columns) * 100 - 2
+      : 0,
+  };
+
+  const zoomableMapSize = showGrid
+    ? {
+        width: mapSize.width,
+        height: mapSize.height,
+      }
+    : {};
```

Comment:
`useMemo`

But also all of the `style` blocks in the JSX will need their own `useMemo`'s too